flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/EvictingWindowOperatorContractTest.java

```diff
@@ -0,0 +1,2654 @@
+	private static <T> void shouldDeleteEventTimeTimerOnElement(Trigger<T, TimeWindow> mockTrigger, final long timestamp) throws Exception {
+		doAnswer(new Answer<TriggerResult>() {
+			@Override
+			public TriggerResult answer(InvocationOnMock invocation) throws Exception {
+				@SuppressWarnings("unchecked")
+				Trigger.TriggerContext context =
+						(Trigger.TriggerContext) invocation.getArguments()[3];
+				context.deleteEventTimeTimer(timestamp);
+				return TriggerResult.CONTINUE;
+			}
+		})
+		.when(mockTrigger).onElement(Matchers.<T>anyObject(), anyLong(), anyTimeWindow(), anyTriggerContext());
+	}
+
+	private static <T> void shouldRegisterProcessingTimeTimerOnElement(Trigger<T, TimeWindow> mockTrigger, final long timestamp) throws Exception {
+		doAnswer(new Answer<TriggerResult>() {
+			@Override
+			public TriggerResult answer(InvocationOnMock invocation) throws Exception {
+				@SuppressWarnings("unchecked")
+				Trigger.TriggerContext context =
+						(Trigger.TriggerContext) invocation.getArguments()[3];
+				context.registerProcessingTimeTimer(timestamp);
+				return TriggerResult.CONTINUE;
+			}
+		})
+				.when(mockTrigger).onElement(Matchers.<T>anyObject(), anyLong(), anyTimeWindow(), anyTriggerContext());
+	}
+
+	private static <T> void shouldDeleteProcessingTimeTimerOnElement(Trigger<T, TimeWindow> mockTrigger, final long timestamp) throws Exception {
+		doAnswer(new Answer<TriggerResult>() {
+			@Override
+			public TriggerResult answer(InvocationOnMock invocation) throws Exception {
+				@SuppressWarnings("unchecked")
+				Trigger.TriggerContext context =
+						(Trigger.TriggerContext) invocation.getArguments()[3];
+				context.deleteProcessingTimeTimer(timestamp);
+				return TriggerResult.CONTINUE;
+			}
+		})
+				.when(mockTrigger).onElement(Matchers.<T>anyObject(), anyLong(), anyTimeWindow(), anyTriggerContext());
+	}
+
+	@SuppressWarnings("unchecked")
+	private static <T, W extends Window> void shouldMergeWindows(final MergingWindowAssigner<T, W> assigner, final Collection<? extends W> expectedWindows, final Collection<W> toMerge, final W mergeResult) {
+		doAnswer(new Answer<Object>() {
+			@Override
+			public Object answer(InvocationOnMock invocation) throws Exception {
+				Collection<W> windows = (Collection<W>) invocation.getArguments()[0];
+
+				MergingWindowAssigner.MergeCallback callback = (MergingWindowAssigner.MergeCallback) invocation.getArguments()[1];
+
+				// verify the expected windows
+				assertThat(windows, containsInAnyOrder(expectedWindows.toArray()));
+
+				callback.merge(toMerge, mergeResult);
+				return null;
+			}
+		})
+				.when(assigner).mergeWindows(anyCollection(), Matchers.<MergingWindowAssigner.MergeCallback>anyObject());
+	}
+
+	private static <T> void shouldContinueOnElement(Trigger<T, TimeWindow> mockTrigger) throws Exception {
+		when(mockTrigger.onElement(Matchers.<T>anyObject(), anyLong(), Matchers.<TimeWindow>any(), anyTriggerContext())).thenReturn(TriggerResult.CONTINUE);
```

Comment:
can we resuse `anyTimeWindow()` to replace `Matchers.<TimeWindow>any()`, just because we've defined the method in this class anyways :-)?